SERVFAIL with TLS

Corefile config
```
.:5305 { 
    fanout . tls://9.9.9.9 {
       tls-server dns.quad9.net
    }
}
 
.:5306 { 
    fanout . 1.1.1.1
}

.:5307 { 
    forward . tls://9.9.9.9 {
       tls_servername dns.quad9.net
    }
}
```

Result fanout with tls: not working
```
$ drill archlinux.org @127.0.0.1 -p 5305
;; ->>HEADER<<- opcode: QUERY, rcode: SERVFAIL, id: 38154
;; flags: qr rd ; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0 
;; QUESTION SECTION:
;; archlinux.org.       IN      A

;; ANSWER SECTION:

;; AUTHORITY SECTION:

;; ADDITIONAL SECTION:

;; Query time: 301 msec
;; SERVER: 127.0.0.1
;; WHEN: Wed Oct 13 22:38:23 2021
;; MSG SIZE  rcvd: 31
```
With logs and errors enabled:
```
Oct 13 22:42:32 name coredns[1202070]: CoreDNS-1.8.6
Oct 13 22:42:32 name coredns[1202070]: linux/amd64, go1.17.1, 796d2e4
Oct 13 22:42:45 name coredns[1202070]: [INFO] 127.0.0.1:37213 - 33130 "A IN archlinux.org. udp 31 false 512" - - 0 0.300800212s
Oct 13 22:42:45 name coredns[1202070]: [ERROR] plugin/errors: 2 archlinux.org. A: attempt limit has been reached
```

Result fanout without tls: works
```
$ drill archlinux.org @127.0.0.1 -p 5306
;; ->>HEADER<<- opcode: QUERY, rcode: NOERROR, id: 37250
;; flags: qr rd ra ; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0 
;; QUESTION SECTION:
;; archlinux.org.       IN      A

;; ANSWER SECTION:
archlinux.org.  1759    IN      A       95.217.163.246

;; AUTHORITY SECTION:

;; ADDITIONAL SECTION:

;; Query time: 6 msec
;; SERVER: 127.0.0.1
;; WHEN: Wed Oct 13 22:40:20 2021
;; MSG SIZE  rcvd: 60
```

Result forward with tls: works
```
$ drill archlinux.org @127.0.0.1 -p 5307
;; ->>HEADER<<- opcode: QUERY, rcode: NOERROR, id: 26425
;; flags: qr rd ra ; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0 
;; QUESTION SECTION:
;; archlinux.org.       IN      A

;; ANSWER SECTION:
archlinux.org.  35259   IN      A       95.217.163.246

;; AUTHORITY SECTION:

;; ADDITIONAL SECTION:

;; Query time: 71 msec
;; SERVER: 127.0.0.1
;; WHEN: Wed Oct 13 22:40:56 2021
;; MSG SIZE  rcvd: 60
```

```
Oct 17 19:27:42 name coredns[176385]: [ERROR] plugin/errors: 2 archlinux.org. A: attempt limit has been reached, last err: dial tcp-tls: unknown network tcp-tls
```

--- a/client.go
+++ b/client.go
@@ -54,7 +54,7 @@ func NewClient(addr, net string) Client {
 // SetTLSConfig sets tls config for client
 func (c *client) SetTLSConfig(cfg *tls.Config) {
 	if cfg != nil {
-		c.net = tcptlc
+		c.net = tcptls
 	}
 	c.transport.SetTLSConfig(cfg)
 }

--- a/const.go
+++ b/const.go
@@ -26,7 +26,7 @@ const (
 	defaultTimeout = 30 * time.Second
 	readTimeout    = 2 * time.Second
 	attemptDelay   = time.Millisecond * 100
-	tcptlc         = "tcp-tls"
+	tcptls         = "tcp-tls"
 	tcp            = "tcp"
 	udp            = "udp"
 )

--- a/setup.go
+++ b/setup.go
@@ -263,7 +263,7 @@ func parseProtocol(f *Fanout, c *caddyfile.Dispenser) error {
 		return c.ArgErr()
 	}
 	net := strings.ToLower(c.Val())
-	if net != tcp && net != udp && net != tcptlc {
+	if net != tcp && net != udp && net != tcptls {
 		return errors.New("unknown network protocol")
 	}
 	f.net = net

--- a/transport.go
+++ b/transport.go
@@ -79,7 +79,7 @@ func (t *transportImpl) dial(ctx context.Context, c *dns.Client) (*dns.Conn, err
 	var conn = new(dns.Conn)
 	var err error
 	if network == tcptlc {
-		conn.Conn, err = tls.DialWithDialer(&d, network, t.addr, c.TLSConfig)
+		conn.Conn, err = tls.DialWithDialer(&d, "tcp", t.addr, c.TLSConfig)
 	} else {
 		conn.Conn, err = d.DialContext(ctx, network, t.addr)
 	}

--- a/transport.go
+++ b/transport.go
@@ -51,9 +51,9 @@ func (t *transportImpl) SetTLSConfig(c *tls.Config) {
 // Dial dials the address configured in transportImpl, potentially reusing a connection or creating a new one.
 func (t *transportImpl) Dial(ctx context.Context, network string) (*dns.Conn, error) {
 	if t.tlsConfig != nil {
-		network = tcptlc
+		network = tcptls
 	}
-	if network == tcptlc {
+	if network == tcptls {
 		return t.dial(ctx, &dns.Client{Net: network, Dialer: &net.Dialer{Timeout: maxTimeout}, TLSConfig: t.tlsConfig})
 	}
 	return t.dial(ctx, &dns.Client{Net: network, Dialer: &net.Dialer{Timeout: maxTimeout}})
@@ -78,7 +78,7 @@ func (t *transportImpl) dial(ctx context.Context, c *dns.Client) (*dns.Conn, err
 	}
 	var conn = new(dns.Conn)
 	var err error
-	if network == tcptlc {
+	if network == tcptls {
 		conn.Conn, err = tls.DialWithDialer(&d, "tcp", t.addr, c.TLSConfig)
 	} else {
 		conn.Conn, err = d.DialContext(ctx, network, t.addr)

